### PR TITLE
Preflight task in main.yml fails in case of RedHat distribution.

### PR DESCRIPTION
**Pull request by mto79**
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

The preflight task in tasks/main.yml fails in case of RedHat/ RHEL distributions. 

The condition:  (ansible_distribution+" "+ansible_distribution_major_version not in checkmk_server_server_stable_os)
compares to: RHEL 7 or RHEL 8 from the variable checkmk_server_server_stable_os. 

But ansible_distribution gives RedHat and not RHEL so the preflight fails.
See below when getting the ansible_distribution
```
ansible localhost -m "setup" | grep distribution
        "ansible_distribution": "RedHat",
```

Issue Number: `e.g. #1`

## What is the new behavior?

- variable checkmk_server_server_stable_os adjusted RHEL 7 => RedHat 7/ RHEL 8 => RedHat 8.
- checkmk_server_server_stable_os -> usage of qoutes in list
- replaced ansible_distribution => ansible_facts.distribution (This the better way nowadays)

## Other information

Trying to do better in description, hopefully it makes sense.

### CODE REVIEW

**Comment by mto79** [comment]
@diademiemi can you have also a look? And @robin-tribe29 i see that i delivered three commits .. have to do better my self in branching .. more a gitlab guy (thee i can squash them latter)

**Comment by mto79** [comment]
And giving ansible and jinja version information .. 

ansible [core 2.12.2]
  config file = ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  ansible collection location = /root/ansible/projects/project_sligro_vms/collections
  executable location = /usr/bin/ansible
  python version = 3.8.12 (default, Sep 16 2021, 10:46:05) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
  jinja version = 2.11.3
  libyaml = True

**Comment by diademiemi** [comment]
I have not had this task fail and I have always tested on CentOS 7 & 8 Stream. How are you getting this error?
I see you are branching this from the main branch, and attempting to merge into the main branch.

When developing for the collection, you should always develop against the devel branch.

The quotes are not needed in the list, and as far as I know the correct naming for this is RHEL, not Red Hat

**Comment by mto79** [comment]
> 



> I have not had this task fail and I have always tested on CentOS 7 & 8 Stream. How are you getting this error? I see you are branching this from the main branch, and attempting to merge into the main branch.
> 
> When developing for the collection, you should always develop against the devel branch.
> 
> The quotes are not needed in the list, and as far as I know the correct naming for this is RHEL, not Red Hat

Thanks stupid of mine to merge against the main branch will do better than that.
Quotes are not necessary but good practice with strings.

I think that it has something to do with  the possible higher ansible version. 

On RHEL 8.6 i get with ansible [core 2.12.2] and jinja version = 2.11.3.

 ansible localhost -m "setup" | grep ansible_distribution
        "ansible_distribution": "RedHat",

**Comment by robin-tribe29** [comment]
Hi @mto79 and don't worry about minor mistakes, nobody is perfect! :)
I will do a review later, but can you update your branch with our `devel` branch? It seems your copy is outdated.
Also, no worries about the commits, we can squash and merge here too. :v:

File: roles/server/defaults/main.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ---
 checkmk_server_server_stable_os:
-  - Debian 11
-  - Ubuntu 20
-  - CentOS 7
-  - CentOS 8
-  - RHEL 7
-  - RHEL 8
```
Comment (robin-tribe29) [inline roles/server/defaults/main.yml:8]: Can we add both versions, until we are certain which way to go?
I see no harm in that, but I do not want to break existing setups.

File: roles/server/tasks/main.yml [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 - name: "Preflight - Fail if target host is not stable for this Role."
   ansible.builtin.fail:
-    msg: "This role is not stable for the target operating system, {{ ansible_distribution }} {{ ansible_distribution_major_version }}."
-  when: (ansible_distribution is not defined) or
-        (ansible_distribution_version is not defined) or
-        (ansible_distribution+" "+ansible_distribution_major_version not in checkmk_server_server_stable_os)
+    msg: "This role is not stable for the target operating system, {{ ansible_facts.distribution }} {{ ansible_facts.distribution_major_version }}."
+  when: (ansible_facts.distribution is not defined) or
+        (ansible_facts.distribution_version is not defined) or
+        (ansible_facts.distribution+" "+ansible_facts.distribution_major_version not in checkmk_server_server_stable_os)
```
Comment (robin-tribe29) [inline roles/server/tasks/main.yml:7]: Can you elaborate on this change?
Is this mandatory moving forward?
What are the up- and downsides?

File: roles/server/defaults/main.yml [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+  - "Debian 11"
+  - "Ubuntu 20"
+  - "CentOS 7"
+  - "CentOS 8"
```
Comment (robin-tribe29) [inline roles/server/defaults/main.yml:6]: I have no strong feelings about the quotes, but we try to avoid changes, that are only cosmetic.

**Review by robin-tribe29 — CHANGES_REQUESTED** [review]

File: roles/server/defaults/main.yml (hunk 3 above)
Comment (mto79) [reply roles/server/defaults/main.yml:6]: It is purely cosmetic but keep in mind that ansible will be a bit stricter in the future about type casting. Eg variabels that are part of a template than its jinja which does the interpretation. (Python)

File: roles/server/tasks/main.yml (hunk 2 above)
Comment (mto79) [reply roles/server/tasks/main.yml:7]: No this more the new way .. would leave it as it is .. but in higher versions is see on the short naming deprecation warnings.

File: roles/server/defaults/main.yml (hunk 1 above)
Comment (mto79) [reply roles/server/defaults/main.yml:8]: Good idea!!

File: roles/server/tasks/main.yml (hunk 2 above)
Comment (diademiemi) [reply roles/server/tasks/main.yml:7]: Are we sure this still works on older versions of Ansible?

Comment (proxy-mto79) [reply roles/server/tasks/main.yml:7]: Well for sure from 2.9.13 .. see discussion as in https://opendev.org/openstack/openstack-ansible-os_nova/commit/b6f03470c4691d31bfc1f0288c68537857805f88

**Comment by proxy-mto79** [comment]
I have the develop branch .. but had to kill my fork .. which closed PR ... pfff github noob

**Comment by robin-tribe29** [comment]
No worries @mto79, all is good.
Will you create a new PR with the changes mentioned in here?

**Comment by mto79** [comment]
Thanx, Robin! i Will make a new PR for you this afternoon. Have the devel now 😀

**Pull request closed without merging**